drivers/opt3001/opt3001.c

```diff
@@ -0,0 +1,179 @@
+    reg &= ~OPT3001_REGS_CONFIG_ME;
+    reg &= ~OPT3001_REGS_CONFIG_FC_MASK;
+
+    reg = htons(reg);
+
+    if (i2c_write_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_init: Error setting device configuration\n");
+        return -OPT3001_ERROR_BUS;
+    }
+    i2c_release(DEV_I2C);
+
+    return OPT3001_OK;
+}
+
+int opt3001_reset(const opt3001_t *dev)
+{
+    uint16_t reg = OPT3001_CONFIG_RESET;
+    reg |= OPT3001_REGS_CONFIG_CT(OPT3001_CONVERSION_TIME);
+    reg = htons(reg);
+
+    /* Acquire exclusive access to the bus. */
+    i2c_acquire(DEV_I2C);
+    if (i2c_write_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_reset: Error setting device configuration\n");
+        return -OPT3001_ERROR_BUS;
+    }
+    i2c_release(DEV_I2C);
+    return OPT3001_OK;
+}
+
+int opt3001_set_active(const opt3001_t *dev)
+{
+    uint16_t reg;
+
+    i2c_acquire(DEV_I2C);
+
+    if (i2c_read_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_set_active: Error reading BUS!\n");
+        return -OPT3001_ERROR_BUS;
+    }
+
+    reg = htons(reg);
+    reg |= OPT3001_REGS_CONFIG_MOC(OPT3001_CONFIG_M_CONTINUOUS);
+
+    reg = htons(reg);
+    if (i2c_write_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_set_active: Error setting device configuration\n");
+        return -OPT3001_ERROR_BUS;
+    }
+
+    i2c_release(DEV_I2C);
+    return OPT3001_OK;
+}
+
+int opt3001_read_lux(const opt3001_t *dev, int32_t *convl)
+{
+    uint8_t exponent;
+    uint16_t reg, mantissa;
+    uint32_t conversion_time = 0;
+
+    i2c_acquire(DEV_I2C);
+
+    while (1) {
+        if (conversion_time >= OPT3001_CONVERSION_TIME_COMBINED) {
+            i2c_release(DEV_I2C);
+            LOG_ERROR("opt3001_read: Conversion in progress!\n");
+            return -OPT3001_ERROR;
+        }
+
+        if (i2c_read_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_CONFIG, &reg, 2, 0) < 0) {
+            i2c_release(DEV_I2C);
+            LOG_ERROR("opt3001_read: Error reading BUS!\n");
+            return -OPT3001_ERROR_BUS;
+        }
+
+        if (htons(reg) & OPT3001_REGS_CONFIG_CRF) {
+            break;
+        }
+
+        xtimer_usleep(1000);
+        conversion_time += 1000;
+    }
+
+    if (i2c_read_regs(DEV_I2C, DEV_ADDR, OPT3001_REGS_RESULT, &reg, 2, 0) < 0) {
+        i2c_release(DEV_I2C);
+        LOG_ERROR("opt3001_read: Error reading BUS!\n");
+        return -OPT3001_ERROR_BUS;
+    }
+
+    exponent = OPT3001_REGS_REG_EXPONENT(htons(reg));
+    mantissa = OPT3001_REGS_REG_MANTISSA(htons(reg));
+
+    *convl = (1 << exponent) * mantissa * 10;
+
+    i2c_release(DEV_I2C);
+    return OPT3001_OK;
+}
+
+int opt3001_read_lux_saul(const opt3001_t *dev, int16_t *convl)
```

Comment:
Since this is only used by saul, you can put that logic into `opt3001_saul.c`.
It only uses public functions anyway.